eolearn/core/eodata_io.py

```diff
@@ -361,11 +370,23 @@ def _load_features(
 def _get_feature_io(
     ftype: FeatureType, path: str, filesystem: FS, temporal_selection: None | slice | list[int] | list[bool]
 ) -> FeatureIO:
+    if ftype is FeatureType.META_INFO:
+        return FeatureIOJson(path, filesystem)
+    if ftype.is_vector():
+        return FeatureIOGeoDf(path, filesystem)
+
     use_zarr = path.endswith(FeatureIOZarr.get_file_extension())
-    constructor = _get_feature_io_constructor(ftype, use_zarr)
+
     if ftype.is_temporal():
-        return constructor(path, filesystem, temporal_selection)  # type: ignore[call-arg]
-    return constructor(path, filesystem)
+        if use_zarr:
+            return FeatureIOZarr(path, filesystem, temporal_selection)
+        if temporal_selection is None:
+            return FeatureIONumpy(path, filesystem)
+        raise IOError(
+            f"Cannot perform loading with temporal selection for numpy data at {path}. Resave feature with"
+            " `use_zarr=True` to enable loading with temporal selections."
+        )
+    return (FeatureIOZarr if use_zarr else FeatureIONumpy)(path, filesystem)
```

Comment:
- Do we support zarr for timeless features?

Reply:
yeah